src/app.provider.ts

```diff
@@ -1,8 +1,13 @@
-import { INestApplication, VersioningType } from '@nestjs/common';
+import {
+  DynamicModule,
+  INestApplication,
+  VersioningType,
+} from '@nestjs/common';
 import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger';
 import { NestFactory } from '@nestjs/core';
 import { IConfigurationService } from '@/config/configuration.service.interface';
 import { json } from 'express';
+import { TestingModule } from '@nestjs/testing';
```

Comment:
This dependency is not available in the final image. We should avoid having references to testing utilities in production code.